net/bridge.go

```diff
@@ -108,3 +199,307 @@ func FindBridgeIP(bridgeName string, subnet *net.IPNet) (net.IP, error) {
 	// No subnet, or none in the required subnet; just return the first one
 	return netdev.CIDRs[0].IP, nil
 }
+
+type BridgeConfig struct {
+	DockerBridgeName string
+	WeaveBridgeName  string
+	DatapathName     string
+	NoFastdp         bool
+	NoBridgedFastdp  bool
+	AWSVPC           bool
+	NPC              bool
+	MTU              int
+	Mac              string
+	Port             int
+}
+
+func CreateBridge(procPath string, config *BridgeConfig) (Bridge, error) {
+	bridgeType, err := DetectBridgeType(config.WeaveBridgeName, config.DatapathName)
+	if err != nil {
+		return nil, err
+	}
+
+	if bridgeType == nil {
+		bridgeType = bridgeImpl{}
+		if !config.NoFastdp {
+			bridgeType = bridgedFastdpImpl{}
+			if config.NoBridgedFastdp {
+				bridgeType = fastdpImpl{}
+				config.DatapathName = config.WeaveBridgeName
+			}
+			odpSupported, err := odp.CreateDatapath(config.DatapathName)
+			if err != nil {
+				return nil, errors.Wrapf(err, "creating datapath %q", config.DatapathName)
+			}
+			if !odpSupported {
+				bridgeType = bridgeImpl{}
+			}
+		}
+		if err := bridgeType.init(config); err != nil {
+			return nil, err
+		}
+		if err := configureIPTables(config); err != nil {
+			return bridgeType, errors.Wrap(err, "configuring iptables")
+		}
+	}
+
+	if config.AWSVPC {
+		// Set proxy_arp on the bridge, so that it could accept packets destined
+		// to containers within the same subnet but running on remote hosts.
+		// Without it, exact routes on each container are required.
+		if err := sysctl(procPath, "net/ipv4/conf/"+config.WeaveBridgeName+"/proxy_arp", "1"); err != nil {
+			return bridgeType, errors.Wrap(err, "setting proxy_arp")
+		}
+		// Avoid delaying the first ARP request. Also, setting it to 0 avoids
+		// placing the request into a bounded queue as it can be seen:
+		// https://git.kernel.org/cgit/linux/kernel/git/stable/linux-stable.git/tree/net/ipv4/arp.c?id=refs/tags/v4.6.1#n819
+		if err := sysctl(procPath, "net/ipv4/neigh/"+config.WeaveBridgeName+"/proxy_delay", "0"); err != nil {
+			return bridgeType, errors.Wrap(err, "setting proxy_arp")
+		}
+	}
+
+	if err := linkSetUpByName(config.WeaveBridgeName); err != nil {
+		return bridgeType, err
+	}
+
+	if err := ConfigureARPCache(procPath, config.WeaveBridgeName); err != nil {
+		return bridgeType, errors.Wrapf(err, "configuring ARP cache on bridge %q", config.WeaveBridgeName)
+	}
+
+	return bridgeType, nil
+}
+
+func (b bridgeImpl) initPrep(config *BridgeConfig) error {
+	mac, err := net.ParseMAC(config.Mac)
+	if err != nil {
+		return errors.Wrapf(err, "parsing bridge MAC %q", config.Mac)
+	}
+
+	linkAttrs := netlink.NewLinkAttrs()
+	linkAttrs.Name = config.WeaveBridgeName
+	linkAttrs.HardwareAddr = mac
+	if config.MTU == 0 {
+		config.MTU = 65535
```

Comment:
1. 0 just means nobody has set it yet.
2. Yes, packet length is a 16-bit value.
3. Sleeve has its own PMTU discovery, and will fragment packets that are too big to transport on the underlying network. However the PMTU value is per path, and we only get to set one MTU on the bridge and on all veths, for all paths, so we go for the biggest possible value.
Note this code is only hit when not using fastdp.  Fastdp cannot deal with fragments, so we have to set the MTU up front, small enough to work on all paths.